### PR TITLE
[feature] Add custom spark id generator option. Closes #551

### LINKED ISSUE #551: Feature Request: Ability to hook up a custom spark ID generator
I noticed that Primus uses [yeast](https://github.com/unshiftio/yeast) in the event that the transformer doesn't provide IDs. In our particular case, we're using the [ws](https://github.com/websockets/ws) implementation, and we'd like to use our own custom ID generator due to particular application requirements. We could, of course, create a mapping of spark IDs to our own generated IDs but that gets messy. It would be nice if we could just plug in our generator into Primus and use that ID.

Is there an easy way to accomplish this currently?

### PR DESCRIPTION
**Pull request by nanowizard**
Signed-off-by: Ryan Sivek <nanowizardry@gmail.com>

### CODE REVIEW

File: test/primus.test.js [hunk 1]
```diff
@@ -104,6 +104,22 @@ describe('Primus', function () {
 
     throw new Error('I should have throwed above');
   });
+  
+  it('accepts a third-party spark id generator', function () {
```
Comment (lpinca) [inline test/primus.test.js]: Can you please move this test to `spark.test.js`?

File: test/primus.test.js [hunk 2]
```diff
@@ -104,6 +104,22 @@ describe('Primus', function () {
...
+    function testIdGenerator() {
+      return 'test_'+(Math.random()+1).toString(36).substring(7);
```
Comment (lpinca) [inline test/primus.test.js]: Can't we simply return a dummy string like `'foo'` and create only a spark below?
I think it's sufficient as all we need to test is that the function is invoked and the spark id is set to the return value of this function.

File: test/primus.test.js (hunk 1 above)
Comment (nanowizard) [inline test/primus.test.js]: Sure thing

File: test/primus.test.js (hunk 2 above)
Comment (nanowizard) [inline test/primus.test.js]: Yes, that's probably sufficient. I'll push a new commit momentarily with your recommended changes.

**Comment by lpinca** [comment]
Thank you!

**Pull request merged**